framework_tvb/tvb/core/entities/file/files_update_manager.py

```diff
@@ -195,17 +226,34 @@ def run_all_updates(self):
             no_ignored = 0
             start_time = datetime.now()
 
-            # Read DataTypes in pages to limit the memory consumption
-            for current_idx in range(0, total_count, self.DATA_TYPES_PAGE_SIZE):
-                datatypes_for_page = dao.get_all_datatypes(current_idx, self.DATA_TYPES_PAGE_SIZE)
-                count_ok, count_error, count_ignored = self.__upgrade_datatype_list(datatypes_for_page)
-                no_ok += count_ok
-                no_error += count_error
-                no_ignored += count_ignored
+            if TvbProfile.current.version.DATA_CHECKED_TO_VERSION < 4:
+                # Read DataTypes in pages to limit the memory consumption
+                for current_idx in range(0, total_count, self.DATA_TYPES_PAGE_SIZE):
+                    datatypes_for_page = dao.get_all_datatypes(current_idx, self.DATA_TYPES_PAGE_SIZE)
+                    count_ok, count_error, count_ignored = self.__upgrade_datatype_list(datatypes_for_page)
+                    no_ok += count_ok
+                    no_error += count_error
+                    no_ignored += count_ignored
+
+                    self.log.info(
+                        "Updated H5 files so far: %d [fine:%d, error:%d, ignored:%d of total:%d, in: %s min]" % (
+                            current_idx + len(datatypes_for_page), no_ok, no_error, no_ignored, total_count,
+                            int((datetime.now() - start_time).seconds / 60)))
+            else:
+                # Migrating the configuration file
+                new_stored_settings = {KEY_ADMIN_DISPLAY_NAME: 'Administrator', KEY_ENABLE_KC_LOGIN: False,
+                                       KEY_KC_WEB_CONFIGURATION: '', KEY_KC_CONFIGURATION: 'add_keycloak_path_here'}
+                manager = TvbProfile.current.manager
+                manager.add_entries_to_config_file(new_stored_settings)
+                manager.delete_entries_from_config_file(['URL_WEB'])
+
+                file_paths = self._get_all_h5_paths()
+                total_count = len(file_paths)
+                no_ok, no_error = self.__upgrade_h5_list(file_paths)
 
-                self.log.info("Updated H5 files so far: %d [fine:%d, error:%d, ignored:%d of total:%d, in: %s min]" % (
-                    current_idx + len(datatypes_for_page), no_ok, no_error, no_ignored, total_count,
-                    int((datetime.now() - start_time).seconds / 60)))
+                self.log.info("Updated H5 files in total: %d [fine:%d, failed:%d in: %s min]" % (
+                    total_count, no_ok, no_error, int((datetime.now() - start_time).seconds / 60)))
+                delete_old_burst_table_after_migration()
```

Comment:
This module was supposed to be generic, for all updates. Now it has hard-coded logic for the latest update.
Also the extra param to method upgrade_file is not generic (can't you use self.something for the extra information?)

Reply:
It can't be totally generic, because the last update is done in a different way than the old ones. There is an if      TvbProfile.current.version.DATA_CHECKED_TO_VERSION < 4 which separates the 2 styles so the previous updates are still working, I even tested them when doing the tvb data update task.

Also the need of that parameter is a little hard to explain in written form. Perhaps we can discuss it in a short call.